src/firebase-namespace.ts

```diff
@@ -273,16 +278,15 @@ export class FirebaseNamespace {
   }
 
   /**
-   * Firebase services available off of a FirebaseNamespace instance. These are monkey-patched via
-   * registerService(), but we need to include a dummy implementation to get TypeScript to
-   * compile it without errors.
+   * Gets the `Auth` service namespace. The returned namespace can be used to get the
+   * `Auth` service for the default app or an explicitly specified app.
    */
-  /* istanbul ignore next */
-  public auth(): FirebaseServiceInterface {
-    throw new FirebaseAppError(
-      AppErrorCodes.INTERNAL_ERROR,
-      'INTERNAL ASSERT FAILED: Firebase auth() service has not been registered.',
-    );
+  get auth(): FirebaseServiceNamespace<Auth> {
```

Comment:
this line breaks [es6 import mocking](https://railsware.com/blog/2017/01/10/mocking-es6-module-import-without-dependency-injection/)..